packages/orga/src/types.ts

```diff
@@ -181,11 +184,37 @@ export interface Link extends Literal {
   search?: string | number;
 }
 
+export type FootnoteRef = FootnoteReference | FootnoteInline | FootnoteAnonymous;
+
 export interface FootnoteReference extends Node {
   type: 'footnote.reference';
   label: string;
 }
 
+export interface FootnoteInline extends Parent {
+  type: 'footnote.inline';
+  label: string;
+  children: PhrasingContent[];
+}
+
+export interface FootnoteAnonymous extends Parent {
+  type: 'footnote.anonymous';
+  children: PhrasingContent[];
+}
+
+export interface FootnoteAnonymousBegin extends Node {
+  type: 'footnote.anonymous.begin';
+}
+
+export interface FootnoteInlineBegin extends Node {
+  type: 'footnote.inline.begin';
+  label: string;
+}
+
+export interface FootnoteReferenceEnd extends Node {
+  type: 'footnote.reference.end';
+}
+
```

Comment:
If you are okay with the above suggestion regarding a single type for footnote reference, do we still need these token types? Off the top of my head, we basically have 2 options for handling this:
1. we can finish `FootnoteReference` parsing within the inline tokenization phase, do all the nested content parsing within the lexer (`src/tokenize/inline.ts`), that way I don't think we need to generate these tokens anymore since there's no work for the parsing part.
2. or we do it the "proper way", streaming out tokens in lexer and let the parser (`src/parse/paragraph.ts`) do the assembling work. in that case we need to move a lot of the logic of inline parsing from the lexer into the parser as well because FootnoreReference now contains `PhrasingContent`. That would be a much bigger change. Maybe we should do it at some point.

Reply:
@xiaoxinghu I think it's already doing the second option - first we lex these tokens, then we handle the nesting in the parser.

Reply:
Note that I added an `ignore` option to the inline lexer, which allows us to ensure that `]` isn't just consumed as a regular character when we are expecting to close a footnote. An alternative to this would be to just spit any `]` out as a token, and then in the parser decide whether to treat it as a regular character or a footnote close. Seems to work fine at the moment though.

Reply:
Okay, I am convinced about this one now. It's probably the best way to handle this without major refactoring. I do have plans to do the inline parsing in a more robust way. Just an FYI the `link` has a similar story, currently, it's pretty barebone.